kernel_module/legion-laptop.c

```diff
@@ -2842,84 +2875,93 @@ struct WMIFanTableRead {
 	u32 FSS7;
 	u32 FSS8;
 	u32 FSS9;
-	u32 FSSA;
+	// Ignored for now
+	u32 FTLE;
+	u32 FST0;
+	u32 FST1;
+	u32 FST2;
+	u32 FST3;
+	u32 FST4;
+	u32 FST5;
+	u32 FST6;
+	u32 FST7;
+	u32 FST8;
+	u32 FST9;
 } __packed;
 
 static ssize_t wmi_read_fancurve_custom(const struct model_config *model,
 					struct fancurve *fancurve)
 {
-	u8 buffer[88];
+	struct WMIFanTableRead fantable;
```

Comment:
Good idea. By adding this new fields we get exactly 88 Bytes. These fields where not defined in the ACPI code of other models.